overlord/ifacestate/helpers.go

```diff
@@ -1102,8 +1103,23 @@ func snapsWithSecurityProfiles(st *state.State) ([]*interfaces.SnapAppSet, error
 				continue
 			}
 
-			// TODO:COMPS: add components to SnapState.PendingSecurity
-			set, err := interfaces.NewSnapAppSet(snapInfo, nil)
+			components := make([]*snap.ComponentInfo, 0, len(snapst.PendingSecurity.Components))
+			for _, csi := range snapst.PendingSecurity.Components {
+				cpi := snap.MinimalComponentContainerPlaceInfo(
+					csi.Component.ComponentName,
+					csi.Revision,
+					instanceName,
+				)
+				container := snapdir.New(cpi.MountDir())
+				ci, err := snap.ReadComponentInfoFromContainer(container, snapInfo, csi)
+				if err != nil {
+					logger.Noticef("cannot read component info for snap %q: %s", instanceName, err)
+					continue
+				}
+				components = append(components, ci)
```

Comment:
this inner loop is not exercized by tests afaict

Reply:
Extending an existing test in 8d333b3dc23d74e3223f11da207bebef7c578aef.